### PR TITLE
feat: ApplicationSet add `create-delete` policy #9101

### PR DESCRIPTION
**Pull request by takumakume**
ref: https://github.com/argoproj/argo-cd/issues/9101#issuecomment-1295192668

Add a `create-delete` policy to create or delete in ApplicationSet.

> My team is having trouble because we are using the Pull Request Generator + Image Updater.
> 
> The `create-only` policy alleviates this problem. However, it is difficult to use because it is not deleted when the pull request is closed.
> 
> I think the root solution is to keep overwriting or 3-way-merge.
> 
> However, this problem has not been solved for a long time because it requires a major modification.
> 
> I propose adding a `create-delete` policy as a way to alleviate this problem quickly.
> 
> I think this would be useful for applications like the Pull request Generator, which often have a short lifecycle.

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Review by crenshaw-dev — COMMENTED** [review]
@takumakume looks great! Would you be up for writing some tests? `TestReconcilerValidationErrorBehaviour` might be a good basis.

**Comment by takumakume** [comment]
@crenshaw-dev

Thanks for the review!

Integration test for [`utils.Policies`](https://github.com/argoproj/argo-cd/blob/master/applicationset/utils/policy.go) was not found, so it has been added. https://github.com/argoproj/argo-cd/pull/11107/commits/f9d6a0301c56280b1c1469c74f2c1d0d787f5bce

File: docs/operator-manual/applicationset/Controlling-Resource-Modification.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ See 'How to modify ApplicationSet container parameters' below for detailed steps
 
 The ApplicationSet controller supports a parameter `--policy`, which is specified on launch (within the controller Deployment container), and which restricts what types of modifications will be made to managed Argo CD `Application` resources.
 
-The `--policy` parameter takes three values: `sync`, `create-only`, and `create-update`. (`sync` is the default, which is used if the `--policy` parameter is not specified; the other policies are described below).
+The `--policy` parameter takes four values: `sync`, `create-only`, `create-update`, and `create-delete`. (`sync` is the default, which is used if the `--policy` parameter is not specified; the other policies are described below).
```
Comment (mubarak-j) [inline docs/operator-manual/applicationset/Controlling-Resource-Modification.md:21]:
```suggestion
The `--policy` parameter takes one of the following valid values: `sync`, `create-only`, `create-update`, and `create-delete`. (`sync` is the default, which is used if the `--policy` parameter is not specified; the other policies are described below).
```

**Comment by takumakume** [comment]
@crenshaw-dev Test code added. Please check.

**Review by crenshaw-dev — APPROVED** [review]

**Pull request merged**